### PR TITLE
Migrate from jclouds to AWS SDK

### PR DESCRIPTION
**Pull request by BradyShober**
Closes #765 
Removes the unmaintained jclouds library and migrates to pure AWS SDK

### Testing done

Tested on a running instance with a pipeline to create artifacts and stashes, verified artifacts can be downloaded, and verified the artifact/stash delete flags.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -292,29 +265,10 @@
         <configuration>
           <rules combine.children="append">
             <requireSameVersions>
-              <dependencies>
-                <dependency>com.google.inject:guice</dependency>
-                <dependency>com.google.inject.extensions:guice-assistedinject</dependency>
-              </dependencies>
+              <dependencies/>
```
Comment (jglick) [inline pom.xml]: Thus this rule could be deleted, and thus the whole `plugin`, and even the whole `build` section.

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/NetworkTest.java [hunk 2]
```diff
@@ -1,414 +0,0 @@
...
-
-/**
- * Explores responses to edge cases such as server errors and hangs.
- * We do mocking at the jclouds level, rather than using (say) S3Mock, because:
- * <ul>
- * <li>We are interested here in the behavior of the generic Jenkins integration code in this package,
- *     as well as its dependencies in Jenkins core and Pipeline, not that of a particular blob store.
- * <li>S3-specific failure modes are of interest but are generally hard to simulate using those mock frameworks anyway.
- *     For example, some S3 mock frameworks completely ignore authentication.
- *     Conversely, some failure modes we want to test may not be supported by an S3 mock framework.
```
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_jclouds/NetworkTest.java:67]: Removing test coverage here. I need to think about how important it is, and whether it would be possible to cover in a different way.

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java [hunk 3]
```diff
@@ -54,35 +54,43 @@
...
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+import software.amazon.awssdk.services.s3.model.ObjectIdentifier;
+import software.amazon.awssdk.services.s3.model.S3Object;
+import software.amazon.awssdk.services.s3.paginators.ListObjectsV2Iterable;
 
 import static io.jenkins.plugins.artifact_manager_jclouds.TikaUtil.detectByTika;
 
 /**
- * Jenkins artifact/stash implementation using any blob store supported by Apache jclouds.
+ * Jenkins artifact/stash implementation using any blob store backed by the AWS SDK.
```
Comment (jglick) [inline src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java:84]: (Could probably be renamed with some XStream tricks.)

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/JCloudsVirtualFileTest.java
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/JCloudsVirtualFileTest.java:1]: Not clear to me why this is being deleted. Seems like it should be possible to preserve with some minor edits to setup code.

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/JCloudsArtifactManagerTest.java [hunk 4]
```diff
@@ -198,9 +192,6 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         wc.getPage(b, "artifact/3/");
         System.err.println("3/4 subdir");
         wc.getPage(b, "artifact/3/4/");
-        int httpCount = httpLogging.getRecords().size();
-        System.err.println("total count: " + httpCount);
-        assertThat(httpCount, lessThanOrEqualTo(13));
```
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/JCloudsArtifactManagerTest.java:203]: Losing a significant assertion here.
